File: ext/RmlUi.Native/src/Debugger.cpp
#include "RmlNative.h"
#include <RmlUi/Debugger.h>

RMLUI_CAPI void rml_Debugger_Initialise(Rml::Context* context) {
	Rml::Debugger::Initialise(context);
}

RMLUI_CAPI void rml_Debugger_Shutdown() {
	Rml::Debugger::Shutdown();
}

RMLUI_CAPI void rml_Debugger_SetContext(Rml::Context* context) {
	Rml::Debugger::SetContext(context);
}

RMLUI_CAPI void rml_Debugger_SetVisible(bool visible) {
	Rml::Debugger::SetVisible(visible);
}

RMLUI_CAPI bool rml_Debugger_IsVisible() {
	return Rml::Debugger::IsVisible();
}

File: ext/RmlUI.Net/Debugger.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RmlUiNet
{
    public static class Debugger
    {
        public static void Initialise(Context context)
        {
            Native.Debugger.Initialise(context.NativePtr);
        }

        public static void SetContext(Context context)
        {
            Native.Debugger.SetContext(context.NativePtr);
        }

        public static bool IsVisible()
        {
            return Native.Debugger.IsVisible();
        }

        public static void SetVisible(bool visible)
        {
            Native.Debugger.SetVisible(visible);
        }

        public static void Shutdown()
        {
            Native.Debugger.Shutdown();
        }
    }
}
